File: funcscript/parser/syntax/FuncScriptParser.GetSpaceLessString.cs
using FuncScript.Model;
namespace FuncScript.Core
{
    public partial class FuncScriptParser
    {
        record GetSpaceLessStringResult(string Text, ParseNode ParseNode, int NextIndex)
            : ParseResult(ParseNode, NextIndex);

        static GetSpaceLessStringResult GetSpaceLessString(ParseContext context, int index)
        {
            if (index >= context.Expression.Length)
                return new GetSpaceLessStringResult(null, null, index);

            var i = index;

            if (i >= context.Expression.Length || IsCharWhiteSpace(context.Expression[i]))
                return new GetSpaceLessStringResult(null, null, index);

            i++;
            while (i < context.Expression.Length && !IsCharWhiteSpace(context.Expression[i]))
                i++;

            var text = context.Expression.Substring(index, i - index);
            var parseNode = new ParseNode(ParseNodeType.Identifier, index, i - index);
            return new GetSpaceLessStringResult(text, parseNode, i);
        }
    }
}

File: ts/core/src/parser/syntax/FuncScriptParser.GetSpaceLessString.ts
import { ParseContext, ParseNode, ParseNodeType, ParseResult } from "../FuncScriptParser.Main";
import { isCharWhiteSpace } from "../FuncScriptParser.Utils";

export function GetSpaceLessString(context: ParseContext, index: number): ParseResult {
    if (index >= context.Expression.length) {
        return { ParseNode: null, NextIndex: index };
    }

    let i = index;

    // Assuming a placeholder for isCharWhiteSpace as it's not to be implemented
    if (i >= context.Expression.length || isCharWhiteSpace(context.Expression[i])) {
        return { ParseNode: null, NextIndex: index };
    }

    i++;
    while (i < context.Expression.length && !isCharWhiteSpace(context.Expression[i])) {
        i++;
    }

    const text = context.Expression.substring(index, i);
    const parseNode = new ParseNode(ParseNodeType.Identifier, index, i - index);
    return { ParseNode: parseNode, NextIndex: i };
}